hashicorp/web-unified-docs
# feat: add PRODUCT_CONFIG

### Description

[🎟️ Asana task](https://app.asana.com/0/1207899860738460/1208799860712577/f)

This PR creates a PRODUCT_CONFIG object that can be used across the repo to access any product config information. This mainly gets used in the migration scripts but also is used across a few other scripts in the repo. I also added the `productSlug` attribute to contain the product name that the repo belongs to. For example, terraform-docs-agents content shows up in the terraform section of dev-portal so the `productSlug` for that section is 'terraform'.

### Testing
- All-docs-paths: go to [/api/all-docs-paths](https://web-unified-docs-av1aihixt-hashicorp.vercel.app/api/all-docs-paths) and verify that the paths are working
- content/[productSlug]/doc/[version]/[...docsPath]: go to [example](https://web-unified-docs-av1aihixt-hashicorp.vercel.app/api/content/terraform-cdk/doc/v0.15.x/cdktf/cli-reference/cli-configuration) or other pages and verify that content gets returned as expected
- add-version-to-internal-links: go to [example](https://web-unified-docs-av1aihixt-hashicorp.vercel.app/api/content/terraform-cdk/doc/v0.15.x/cdktf/cli-reference/cli-configuration) and verify that links include the version (v0.15.x)
- build-target-repos: check out this branch and run the command below. Verify that it works as expected with no errors
```
node ./scripts/migrate-content/migrate-content.mjs terraform 
``` 
- clone-all-repos: check out this branch and run the command below. Verify that `.content-source-repos` directory is created and contains docs repos
```
node ./scripts/migrate-content/clone-all-repos.mjs 
``` 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208799860712577
